algorithms/linfa-ica/benches/fast_ica.rs

```diff
@@ -0,0 +1,60 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use linfa::{dataset::DatasetBase, traits::Fit};
+use linfa_ica::fast_ica::{FastIca, GFunc};
+use ndarray::{array, concatenate};
+use ndarray::{Array, Array2, Axis};
+use ndarray_rand::{rand::SeedableRng, rand_distr::Uniform, RandomExt};
+use rand_xoshiro::Xoshiro256Plus;
+
+fn perform_ica(size: usize) -> () {
+    let sources_mixed = create_data(size);
+
+    let ica = FastIca::params().gfunc(GFunc::Logcosh(1.0));
```

Comment:
Also run separate benchmarks with `GFunc::Exp` and `GFunc::Cube`.

Also set `random_seed` for reproducible results.

Reply:
lines 38-40 have the random_seed set to add reproducible noise. The original signals are created from 

`Array::linspace(0., 8., nsamples)` which should guarantee reproducibility. 

I'm adding the additional functions now though.

Reply:
By `random_seed` I meant calling `FastIca::params().random_seed()` to set a reproducible seed for the ICA algorithm itself.

Reply:
oh gotcha will do now.